langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -1060,6 +1060,73 @@ function testLangLibFuncWithNamedArgExpr() {
     }
 }
 
+function testEmptyRegexpFind() {
+    // find
+    regexp:Span? resA1 = regexp:find(re = re ``, str = "HelloWorld");
+    assertTrue(resA1 is null);
+    regexp:Span? resA2 = regexp:find(re = re `World`, str = "");
+    assertTrue(resA2 is null);
+    regexp:Span? resA3 = regexp:find(re = re ``, str = "");
+    assertTrue(resA3 is null);
+    regexp:Span? resA4 = regexp:find(re = re `${""}`, str = "");
+    assertTrue(resA4 is null);
+    regexp:Span? resA5 = regexp:find(re = re `${""}`, str = "HelloWorld");
+    assertTrue(resA5 is null);
+    string regexStrA = "";
+    regexp:Span? resA6 = regexp:find(re = re `${regexStrA}`, str = "HelloWorld");
+    assertTrue(resA6 is null);
+    regexp:Span? resA7 = regexp:find(re = re `${regexStrA}`, str = "");
+    assertTrue(resA7 is null);
+    regexp:Span? resA8 = regexp:find(re = re `(.*)`, str = "");
+    assertTrue(resA8 is regexp:Span);
+
+    // find all
+    regexp:Span[] resB1 = regexp:findAll(re ``, "There once was a king who liked to sing");
+    assertEquality(0, resB1.length());
+    regexp:Span[] resB2 = regexp:findAll(re `(\w+ing)`, "");
+    assertEquality(0, resB2.length());
+    regexp:Span[] resB3 = regexp:findAll(re ``, "");
+    assertEquality(0, resB3.length());
+    regexp:Span[] resB4 = regexp:findAll(re `${""}`, "");
+    assertEquality(0, resB4.length());
+    regexp:Span[] resB5 = regexp:findAll(re `${""}`, "There once was a king who liked to sing");
+    assertEquality(0, resB5.length());
+
+   //  // find groups
+    regexp:Groups? resC1 = regexp:findGroups(re ``, "Butter was bought by Betty but the butter was bitter");
+    assertTrue(resC1 is null);
+    regexp:Groups? resC2 = regexp:findGroups(re `(\w+tt\w+)`, "");
+    assertTrue(resC2 is null);
+    regexp:Groups? resC3 = regexp:findGroups(re ``, "");
+    assertTrue(resC3 is null);
+    regexp:Groups? resC4 = regexp:findGroups(re `${""}`, "");
+    assertTrue(resC4 is null);
+
+   //  // find all groups
+    regexp:Groups[] resD1 = regexp:findAllGroups(re ``, "rubble, trouble, bubble, hubble");
+    assertEquality(0, resD1.length());
+    regexp:Groups[] resD2 = regexp:findAllGroups(re `(\w+ble)`, "");
+    assertEquality(0, resD2.length());
+    regexp:Groups[] resD3 = regexp:findAllGroups(re ``, "");
+    assertEquality(0, resD3.length());
+    regexp:Groups[] resD4 = regexp:findAllGroups(re `${""}`, "");
+    assertEquality(0, resD4.length());
+}
+
+function testEmptyRegexpMatch() {
+    // matchAt
+    regexp:Span? resA1 = regexp:matchAt(re = re ``, str = "HelloWorld");
+    assertTrue(resA1 is null);
+    regexp:Span? resA2 = regexp:matchAt(re = re ``, str = "HelloWorld", startIndex = 4);
+    assertTrue(resA2 is null);
+    string regexStrA = "";
+    regexp:Span? resA3 = regexp:matchAt(re = re `${regexStrA}`, str = "HelloWorld");
```

Comment:
Can you check this behaviour with Javascript. Imo, This is incorrect since we define our regex pattern as a empty string

Reply:
The string value of ``re `${regexStrA}` `` and ```re `` ``` both will be evaluated to `""`.
@pcnfernando is there any way to distinguish between the above two cases?

Reply:
As per an offline discussion with @mohanvive @pcnfernando @SasinduDilshara it was decided provide a compile-time error for regexps with empty values ``` re `` ```. 
And empty values which are available at run-time, will be mapped to the non-capturing group `(?:)`.

Will be fixing #39794 as well.